src/mesh/parallel/shiftedmetric.cxx

```diff
@@ -66,35 +66,21 @@ void ShiftedMetric::cachePhases() {
   // As we're attached to a mesh we can expect the z direction to
   // not change once we've been created so precalculate the complex
   // phases used in transformations
-  int nmodes = mesh.LocalNz / 2 + 1;
+  nmodes = mesh.LocalNz / 2 + 1;
   BoutReal zlength = mesh.getCoordinates()->zlength();
 
-  // Allocate storage for our 3d vector structures.
-  // This could be made more succinct but this approach is fairly
-  // verbose --> transparent
-  fromAlignedPhs.resize(mesh.LocalNx);
-  toAlignedPhs.resize(mesh.LocalNx);
-
-  for (int jx = 0; jx < mesh.LocalNx; jx++) {
-    fromAlignedPhs[jx].resize(mesh.LocalNy);
-    toAlignedPhs[jx].resize(mesh.LocalNy);
-
-    for (int jy = 0; jy < mesh.LocalNy; jy++) {
-      fromAlignedPhs[jx][jy].resize(nmodes);
-      toAlignedPhs[jx][jy].resize(nmodes);
-    }
-  }
-
-  // To/From field aligned phases
-  for (int jx = 0; jx < mesh.LocalNx; jx++) {
-    for (int jy = 0; jy < mesh.LocalNy; jy++) {
-      for (int jz = 0; jz < nmodes; jz++) {
-        BoutReal kwave = jz * 2.0 * PI / zlength; // wave number is 1/[rad]
-        fromAlignedPhs[jx][jy][jz] =
-            dcomplex(cos(kwave * zShift(jx, jy)), -sin(kwave * zShift(jx, jy)));
-        toAlignedPhs[jx][jy][jz] =
-            dcomplex(cos(kwave * zShift(jx, jy)), sin(kwave * zShift(jx, jy)));
-      }
+  // Allocate storage for our 3d phase information.
+  fromAlignedPhs = Tensor<dcomplex>(mesh.LocalNx, mesh.LocalNy, nmodes);
+  toAlignedPhs = Tensor<dcomplex>(mesh.LocalNx, mesh.LocalNy, nmodes);
+
+  //To/From field aligned phases
+  BOUT_FOR(i, mesh.getRegion2D("RGN_ALL")) {
+    for(int jz=0;jz<nmodes;jz++) {
+      BoutReal kwave=jz*2.0*PI/zlength; // wave number is 1/[rad]
+      fromAlignedPhs(i.x(), i.y(), jz) =
```

Comment:
Is it worth caching the values of `i.x()`, `i.y()` outside the z-loop? Probably won't make much difference in the constructor, but could be more important in the other methods

(also clang-format please :slightly_smiling_face:)